src/EditorFeatures/Core/Tagging/TaggerUtilities.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Tagging
+{
+    internal static class TaggerUtilities
+    {
+        public static bool SpanEquals(SnapshotSpan? span1, SnapshotSpan? span2, SpanTrackingMode spanTrackingMode)
+        {
+            if (span1 is null && span2 is null)
+                return true;
+
+            if (span1 is null || span2 is null)
+                return false;
```

Comment:
nit: What do you think of this instead?

if (span1 is null || span2 is null)
    return span1 is null && span2 is null;

Reply:
seems reasonable to me.  :)

Reply:
(i'll make this in a followup pr, so it doesn't have to restart this one).